docs/references/config/config.toml.md

```diff
@@ -0,0 +1,2055 @@
+established with a peer, and the node and peer are negotiating its upgrade into
+a secret authenticated connection.
+
+The value `"0s"` is undefined, and it can lead to unexpected behaviour.
+
+### p2p.dial_timeout
+
+Timeout duration for the low-level dialer that connects to the remote address on the TCP network.
+
+```toml
+dial_timeout = "3s"
+```
+
+| Value type          | string (duration) |
+|:--------------------|:------------------|
+| **Possible values** | &gt;= `"0s"`      |
+
+This parameter is the timeout value for dialing on TCP networks. If a hostname is used instead of an IP address and the
+hostname resolves to multiple IP addresses, the timeout is spread over each consecutive dial, such that each is given an
+appropriate fraction of the time to connect.
+
+Setting the value to `"0s"` disables the timeout.
+
+## Mempool
+Mempool allows gathering and broadcasting uncommitted transactions among nodes.
+
+The **mempool** is storage for uncommitted transactions; the **mempool cache** is internal storage within the
+mempool for seen transactions. The mempool cache provides a list of transactions already received to filter out
+incoming duplicate transactions and prevent duplicate full transaction validations.
+
+### mempool.type
+The type of mempool this node will use.
+```toml
+type = "flood"
+```
+
+| Value type          | string    |
+|:--------------------|:----------|
+| **Possible values** | `"flood"` |
+|                     | `"nop"`   |
+
+`"flood"` is the original mempool implemented for CometBFT. It is a concurrent linked list with flooding gossip
+protocol.
+
+`"nop"` is a "no operation" or disabled mempool, where the ABCI application is responsible for storing, disseminating and
+proposing transactions. Note, that it requires empty blocks to be created:
+[`consensus.create_empty_blocks = true`](#consensuscreate_empty_blocks) has to be set.
+
+### mempool.recheck
+Transaction validity check.
+```toml
+recheck = true
+```
+
+| Value type          | boolean |
+|:--------------------|:--------|
+| **Possible values** | `true`  |
+|                     | `false` |
+
+Committing a block affects the application state, hence the remaining transactions in the mempool after a block commit
+might become invalid. Setting `recheck = true` will go through the remaining transactions and remove invalid ones.
+
+### mempool.broadcast
+Broadcast the mempool content (uncommitted transactions) to other nodes.
+```toml
+broadcast = true
+```
+
+| Value type          | boolean |
+|:--------------------|:--------|
+| **Possible values** | `true`  |
+|                     | `false` |
+
+This ensures that uncommitted transactions have a chance to reach multiple validators and get committed by one of them.
+
+Setting this to `false` will stop the mempool from relaying transactions to other peers until they are included in a
+block. Only the peer you send the tx to will see it until it is included in a block.
+
+### mempool.wal_dir
```

Comment:
There seems to be an important bug here... and looks like long standing (I checked on `main` and `v0.34.x` and it exists in both).

The bug is that this config is ignored altogether (!!!). There's _always_ a WAL (i.e., it can't be disabled) in `$CMTHOME/data/cs.wal`.

Reply:
Indeed, no indication that the mempool writes anything to disk.

Actually, `WalDir` and `WalPath` are not not used under `mempool/` or `node/`.

Reply:
We should deprecate and remove.

Reply:
OK, adapted the text to look like `statesync.temp_dir`